receiver/datadogreceiver/internal/translator/sketches.go

```diff
@@ -0,0 +1,266 @@
+		if highestCountProportion > 0 {
+			additionalCount := targetBucketCount - currentAssignedCount
+			if sketchKeys[i] < 0 {
+				negativeBuckets[highestCountIdx] += additionalCount
+			} else {
+				positiveBuckets[highestCountIdx] += additionalCount
+			}
+		}
+
+	}
+
+	return negativeBuckets, positiveBuckets, zeroCount
+}
+
+// convertBucketLayout populates the count for positive or negative buckets in the resulting OTel
+// exponential histogram structure. The bucket layout is dense and consists of an offset, which is the
+// index of the first populated bucket, and a list of counts, which correspond to the counts at the offset
+// bucket's index, and the counts of each bucket after. Unpopulated/empty buckets must be represented with
+// a count of 0. After assigning bucket counts, it sets the offset for the bucket layout
+func convertBucketLayout(inputBuckets map[int]uint64, outputBuckets pmetric.ExponentialHistogramDataPointBuckets) {
+	if len(inputBuckets) == 0 {
+		return
+	}
+	bucketIdxs := make([]int, 0, len(inputBuckets))
+	for k := range inputBuckets {
+		bucketIdxs = append(bucketIdxs, k)
+	}
+	sort.Ints(bucketIdxs)
+
+	bucketsSize := bucketIdxs[len(bucketIdxs)-1] - bucketIdxs[0] + 1 // find total number of buckets needed
+	outputBuckets.BucketCounts().EnsureCapacity(bucketsSize)
+	outputBuckets.BucketCounts().Append(make([]uint64, bucketsSize)...)
+
+	offset := bucketIdxs[0]
+	outputBuckets.SetOffset(int32(offset))
+
+	for _, idx := range bucketIdxs {
+		delta := idx - offset
+		outputBuckets.BucketCounts().SetAt(delta, inputBuckets[idx])
+	}
+}
+
+// getSketchBounds calculates the lower and upper bounds of a sketch bucket based on the index of the bucket.
+// This is based on sketch buckets placing values in bucket so that γ^k <= v < γ^(k+1)
+// See https://github.com/DataDog/datadog-agent/blob/0ada7a97fed6727838a6f4d9c87123d2aafde735/pkg/quantile/config.go#L83
+// and https://github.com/DataDog/sketches-go/blob/8a1961cf57f80fbbe26e7283464fcc01ebf17d5c/ddsketch/ddsketch.go#L468
+func getSketchBounds(index int32) (float64, float64) {
+	if index < 0 {
+		index = -index
+	}
+	return sketchLowerBound(index), sketchLowerBound(index + 1)
+}
+
+// sketchLowerBound calculates the lower bound of a sketch bucket based on the index of the bucket.
+// It uses the index offset and multiplier (represented by (1 / math.Log(gamma))). The logic behind this
+// is based on the DD agent using logarithmic mapping for definition DD agent sketches
+// See:
+// https://github.com/DataDog/opentelemetry-mapping-go/blob/4a6d530273741c84fe2d8f76c55c514cd5eb7488/pkg/quantile/config.go#L54
+// https://github.com/DataDog/sketches-go/blob/8a1961cf57f80fbbe26e7283464fcc01ebf17d5c/ddsketch/mapping/logarithmic_mapping.go#L39
+func sketchLowerBound(index int32) float64 {
+	if index < 0 {
+		index = -index
+	}
+	return math.Exp((float64(index-agentSketchOffset) / (1 / math.Log(gamma))))
```

Comment:
This returns +Inf over an index value of 47096, just copy paste the code bellow into go playground (https://go.dev/play/p/2mnSdR9G7RS).
Causing the  mapSketchBucketsToHistogramBuckets to go into an infinite loop.

```package main

import (
	"fmt"
	"math"
)

func main() {
	relativeAccuracy := 1.0 / 128
	gamma := 1 + 2*relativeAccuracy
	var agentSketchOffset int32 = 1338

	var index int32 = 47096

	lowerBound := math.Exp((float64(index-agentSketchOffset) / (1 / math.Log(gamma))))
	fmt.Printf("%g\n", lowerBound)
}
```

Reply:
resolved